site/content/guides/routing.md

```diff
@@ -0,0 +1,283 @@
+navigating different pages.
+
+Routing can use either modern HTML5 History API to change URL completely or use hash URLs, storing route
+in the URL part after hash mark (#).
+
+## Routing in Frint
+
+To facilitate routing Frint comes with with two packages: `frint-router` and `frint-router-react`.
+
+`frint-router` provides you with router services responsible for storing URL, notifying of URL
+changes and matching it against different path. The services differ in how and where they store URL,
+but provide identical API so you can freely replace them with each other. This package isn't bound
+to any view library.
+
+`frint-router-react` is a React-specific package which complements `frint-router` and makes routing
+really easy if you chose React as your view library by providing some handy components.
+
+### Dependencies
+
+You will need both `frint-router` and `frint-router-react` for this tutorial as well as `frint`,
+`react` and `frint-react`.
+
+```
+$ npm install --save frint-router frint-router-react frint react frint-react
+```
+
+### Creating components
+
+Let's start by creating components. We will have two components for content pages: `HomePage` and
+`AboutPage`. We are also going to have `RootComponent` which will serve as
+container for navigation links and content pages.
+
+We will need to import `React` for creating components as well as `Link` and `Route` components
+from `frint-router-react`.
+
+```js
+import React from 'react';
+import { Link, Route } from 'frint-router-react';
+
+const HomePage = () => <div>Home page</div>;
+const AboutPage = () => <div>About page</div>;
+
+const RootComponent = () => {
+  return (
+    <div>
+      <div className="links">
+        <Link to="/" activeClassName="is-active" exact>[Home]</Link>
+        <Link to="/about" activeClassName="is-active">[About]</Link>
+      </div>
+
+      <div className="content">
+        <Route path="/" component={HomePage} exact />
+        <Route path="/about" component={AboutPage} />
+      </div>
+    </div>
+  );
+};
+```
+
+Let's go over what happens here.
+
+1. `RootComponent` renders section with navigation links using `Link` component. `Link` component
+creates an anchor or a button leading to a certain URL. It also applies CSS class `is-active`
+(passed as `activeClassName` prop) to the link element automatically when current URL matches the
+one passed to the `Link`.
+2. It also makes use of `Route` component to switch between content pages. Whenever `path` passed to
+`Route` component matches current URL it will render `component` from the `props`. Just as with
+`Link` you can choose whether it should be an exact match. You can also pass a Frint app as an `app`
+ prop instead of `component`.
+
+### Import the preferred router
+
+There are three router services available:
+
+- `BrowserRouterService`: uses modern HTML5 History API
+- `HashRouterService`: for legacy browsers
+- `MemoryRouterService`: useful for tests
```

Comment:
it's a bit difficult to decide which one should be used by default by anyone. if your application is backed by a server, then `BrowserRouterService` makes sense.

`BrowserRouterService` would keep URLs in your browser like `/about/page/here`.

But `HashRouterService` can work anywhere, as long as the target page is loaded, and URLs would look like this in the browser `/index.html#/about/page/here`.

And `MemoryRouterService` has nothing to do with the browser's API. it just imitates the behaviour of the other services and very useful in unit testing your app.

We want the developers to pick a service of their own needs when constructing the App, otherwise we end up having too much code in our bundles, that's never executed.

The developer needs to make a conscious decision about it. If you want feature detection, you have to do it yourself when returning the instance of the right service when your `router` provider is computed.